src/lara-plugin/plugin-api/popup.ts

```diff
@@ -104,12 +106,17 @@ export const addPopup = (_options: IPopupOptions): IPopupController => {
   // ACTIVITY_PLAYER_CODE:
   // eslint-disable-next-line prefer-const
   let $dialog: JQuery;
+
   const remove = () => {
     if (options.onRemove) {
       options.onRemove();
     }
     $dialog.remove();
   };
+
+  removeList.push(remove);
+  removeList = removeList.filter((fn) => fn === remove);
```

Comment:
This needs to be inside the remove method so that it removes itself from the list when called, like this:

```
  const remove = () => {
    if (options.onRemove) {
      options.onRemove();
    }
    $dialog.remove();
    removeList = removeList.filter((fn) => fn === remove);
  };
```

Reply:
So close! Thank you for your patience!